src/axom/core/numerics/Matrix.hpp

```diff
@@ -742,7 +742,9 @@ void Matrix< T >::swapColumns( IndexType icol, IndexType jcol )
   T* jcol_data = this->getColumn( jcol );
   for ( IndexType i=0 ; i < m_rows ; ++i )
   {
-    utilities::swap( icol_data[ i ], jcol_data[ i ] );
+    const T temp   = icol_data[ i ];
```

Comment:
Any reason not to use std::swap here? Why do we have utilities::swap?

Reply:
With intel@18 and optimizations turned on, this method was not working properly. 

We have `utilities::swap()` b/c we can host-device decorated, since, `std::swap` may not always work unless the compiler always inlines it.